### PR TITLE
Fix for #16882: Generated CMake build when using tools.build.linker_scripts

### LINKED ISSUE #16882: [bug] Specifying `tools.build.linker_scripts` in my profile results in an invalid command line when linking.
### Describe the bug

I'm using Fedora 40, with conan from the develop2 branch.

While attempting to move my linker_script setup out of cmake and into my conan profile I encountered the error that the way in which the linker command was appended doesn't put a space between the previous linker arguments (if any).

### How to reproduce it

If you create a new cmake_lib and use a profile such as:

```ini
[settings]
os=baremetal
arch=tc162
compiler=gcc
compiler.version=11
compiler.cppstd=20
compiler.libcxx=libstdc++11

[conf]
tools.build:compiler_executables={"c":"tricore-elf-gcc","cpp":"tricore-elf-g++"}
tools.build:linker_scripts=[ "${PROJECT_SOURCE_DIR}/Lcf_Gnuc_Tricore_Tc.ld" ]
tools.cmake.cmake_layout:build_folder=build/tc162
tools.cmake.cmaketoolchain:generator=Ninja Multi-Config

# I have to do this, otherwise zlib tries to use fPIC which isn't supported by tricore.
[options]
*:shared=False
*:fPIC=False
```

Then you'll get an error during the build similar to this:

```
tricore-elf-g++: error: unrecognized argument in option ‘-mtc162-T/home/chipc/.conan2/p/b/mypkg6fa17d873b59e/b/Lcf_Gnuc_Tricore_Tc.ld’
    tricore-elf-g++: note: valid arguments to ‘-mtc’ are: 13 131 16 161 162 18
```

### PR DESCRIPTION
**Pull request by photex**
The generated build otherwise appends the "-Tfoo.ld" argument directly without a preceding space before which can generate an invalid command line.

Changelog: Bugfix: Avoid ``CMakeToolchain`` error when both architecture flags and ``tools.build:linker_scripts`` are defined, due to missing space.
Docs: Omit

Close #16882

- [X] Refer to the issue that supports this Pull Request.
- [ ] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop2/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [ ] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one.

### CODE REVIEW

File: conan/tools/cmake/toolchain/blocks.py [hunk 1]
```diff
@@ -265,7 +265,7 @@ def context(self):
         linker_scripts = [linker_script.replace('\\', '/') for linker_script in linker_scripts]
         linker_scripts = [relativize_path(p, self._conanfile, "${CMAKE_CURRENT_LIST_DIR}")
                           for p in linker_scripts]
-        linker_script_flags = ['-T"' + linker_script + '"' for linker_script in linker_scripts]
+        linker_script_flags = ['-T' + linker_script for linker_script in linker_scripts]
```
Comment (memsharded) [inline conan/tools/cmake/toolchain/blocks.py]: Are you sure these quotes are to be removed? They can be very necessary when there are path containing spaces, can you please try to keep them?

**Review by memsharded — COMMENTED** [review]
Thanks very much for your contribution @photex 

You are right, there is a bug there, and the extra space is necessary.
It would be necessary to add some unit or integration test for this, to make sure this is covered and don't break again in the future. I can also try to help with this.

**Comment by memsharded** [comment]
I have pushed some changes, fixing the test (that test will cover the change already), and trying to keep the original quotes. Can you please test this branch with your build to see if it works? Thanks!

File: conan/tools/cmake/toolchain/blocks.py (hunk 1 above)
Comment (photex) [reply conan/tools/cmake/toolchain/blocks.py]: It was causing cmake warnings previously. But you’re right, there is perhaps another approach to fix those.

**Comment by photex** [comment]
@memsharded you're update fixed the cmake warnings I was seeing as well. :D

File: conan/tools/cmake/toolchain/blocks.py (hunk 1 above)
Comment (photex) [reply conan/tools/cmake/toolchain/blocks.py]: You already found the fix apparently :D

**Pull request merged**